### PR TITLE
User Auth & Monster Routes

### PR DESCRIPTION
**Pull request by villejacob**
Commits from Joe's user authorization branch up to email & password validation

User Routes:
- Get all user info by userID
- Add monster to user by monsterNo
- Rename user's monster w/params
- Add attack to user's monster by monsterNo and attackName

Monster Routes:
- Create new monster w/params
- Create new attack w/params

### CODE REVIEW

File: actions/app.go [hunk 1]
```diff
@@ -38,8 +44,21 @@ func App() *buffalo.App {
...
+
+		authGroup := app.Group("/auth")
+		authGroup.Middleware.Skip(authenticateRequest, createUserHandler, loginHandler) // do not verify a token for these registration/login handlers
+		authGroup.POST("/user", createUserHandler)
+		authGroup.POST("/login", loginHandler)
+
+		app.GET("/user/{userID}", userDataHandler)
+		app.PUT("/user/monster", renameMonsterHandler)
+		app.POST("/user/monster/{monsterNo}", addMonsterHandler)
+		app.POST("/user/monster/{monsterNo}/attack/{attackName}", addMonsterAttackHandler)
```
Comment (villejacob) [inline actions/app.go]: Should these be a group?

File: actions/app.go [hunk 2]
```diff
@@ -38,8 +44,21 @@ func App() *buffalo.App {
...
+		// middleware
+		app.Use(authenticateRequest)
+
+		authGroup := app.Group("/auth")
+		authGroup.Middleware.Skip(authenticateRequest, createUserHandler, loginHandler) // do not verify a token for these registration/login handlers
+		authGroup.POST("/user", createUserHandler)
+		authGroup.POST("/login", loginHandler)
+
+		app.GET("/user/{userID}", userDataHandler)
+		app.PUT("/user/monster", renameMonsterHandler)
```
Comment (villejacob) [inline actions/app.go]: Better way to do this than taking in No and Name as params? Possibly send the monster's _id as the header?

File: actions/app.go (hunk 1 above)
Comment (Joepena) [reply actions/app.go]: Yeah it would be cool to nest them in a "/user" group

File: actions/app.go (hunk 2 above)
Comment (Joepena) [reply actions/app.go]: yeah we can send the monster's _id as a form parameter in the request body

**Pull request closed without merging**